src/core/lightning/add-invoice-for-recipient.ts

```diff
@@ -0,0 +1,24 @@
+import { getWalletFromUsername } from "@core/wallet-factory"
+import { decodeInvoice } from "@domain/ln-invoice"
+
+const addInvoiceForRecipient = async ({ recipient, memo }, { logger }) => {
+  const wallet = await getWalletFromUsername({ username: recipient, logger })
+  const paymentRequest = await wallet.addInvoice({
+    selfGenerated: false,
+    memo,
+  })
+
+  const decodedInvoice = decodeInvoice(paymentRequest)
+
+  if (decodedInvoice instanceof Error) return decodedInvoice
```

Comment:
Shouldn’t we be checking against CoreError?

Reply:
We could if we need to (to something different for `CodeError` for example). Checking against `Error` is kinda the fallback here.